CoffeeShopUI_Angular/src/app/components/sign-up/sign-up.component.ts

```diff
@@ -0,0 +1,88 @@
+;
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { isSubscription } from 'rxjs/internal/Subscription';
+@Component({
+  selector: 'sign-up',
+  templateUrl: './sign-up.component.html',
+  styleUrls: ['./sign-up.component.css'],
+})
+export class SignUpComponent {
+  constructor(private fb: FormBuilder, private http: HttpClient) {}
+  isSubmitted = false;
+ 
+  signUpForm = this.fb.group({
+    firstName: ['', Validators.required],
+    lastName: ['', Validators.required],
+    email: ['', [Validators.required, Validators.email]],
+    password: [
+      '',
+      [
+        Validators.required,
+        // Validators.pattern('(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[.,@$!%*#?&^_-]).{8,99}'),
```

Comment:
Pattern not working with strings for some reason. I suspect the cause is the way Validators.pattern() uses string based parameter patterns. 
Suggestion: use js regex pattern literal or a regexp instance ( look into RegExp class ) 

```suggestion
          Validators.pattern(/(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[.,@$!%*#?&^_-]).{8,99}/),
```